File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    hash = dict()
    
    for cloth in clothes :
        if cloth[1] not in hash :
            hash[cloth[1]] = 1
        else :
            hash[cloth[1]] += 1
    print(hash)
    answer = 1
    for i in hash :
        answer *= ( hash[i] + 1 )
    return answer - 1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;
import java.io.*;
class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        Map<String,List<String>> hash = new HashMap<>();
        
        for(String[] cloth : clothes){
            if(hash.containsKey(cloth[1]))
                hash.get(cloth[1]).add(cloth[0]);
            else{
                hash.put(cloth[1],new ArrayList<>());
                hash.get(cloth[1]).add(cloth[0]);
            }
        }
        
        for(String key : hash.keySet()){
            answer *= (hash.get(key).size() + 1);
        }
        
        return answer - 1;
    }
}